docs/proposals/service-discovery/README.md

```diff
@@ -0,0 +1,171 @@
+---
+title: Service discovery with native Kubernetes naming and resolution
+authors:
+- "@bivas"
+reviewers:
+- "@XiShanYongYe-Chang"
+- TBD
+approvers:
+- "@robot"
+- TBD
+
+creation-date: 2023-06-22
+
+---
+
+# Service discovery with native Kubernetes naming and resolution
+
+## Summary
+
+With the current `ServiceImportController` when a `ServiceImport` object is reconciled, the derived service is prefixed with `derived-` prefix.
+This seems counterintuitive when thinking about service discovery:
+- Assuming the pod is exported as the service `foo`
+- Another pod that wishes to access it on the same cluster will simply call `foo` and Kubernetes will bind to the correct one
+- If that pod is scheduled to another cluster, the original service discovery will fail as there's no service by the name `foo`
+- To find the original pod, the other pod is required to know it is in another cluster and use `derived-foo` to work properly
```

Comment:
By the way, personally I think it's best to use the following approach for multi-cluster services. 


The image is sourced from GKE.

<img width="849" alt="image" src="https://github.com/karmada-io/karmada/assets/17962021/1cd4fb5c-c2a3-461c-b25b-137411353604">

Reply:
youtube video:


https://www.youtube.com/watch?v=MJ0LSvovKqk

Reply:
@chaunceyjiang I actually wish to avoid using `clusterset` so the users will not have to know about `Service` placement and call it locally as with any other `Service` deployed locally